### ISSUE TITLE
Finance Screen numbers go negative

### ISSUE THREAD

**Issue by superchadST345** [OpenLoco/OpenLoco#2363]
**Version information:**
 - Windows 11
 - 24.02.1

**Describe the bug**
At a certain point in the finance screen (in my current game it is December) my train revenue goes negative (-$4 billion, should be positive) then the remainder of the profit also goes negative (-$2 billion) total profit should be close to $6 billion ($4 billion + $2 billion)

pretty sure this is an original bug, it does not lead to actually loosing money, cash still increases, just financial records

i am assuming it happens because revenue reaches a certain point leading to it becoming an unsigned integer or something like that.

**To Reproduce**
Steps to reproduce the behaviour:
1. build large company
2. get to late game (for me was year 2033)
3. train revenue and total profit turn negative

**Expected behaviour**
train revenue should never be negative

**Screenshots**
![image](https://github.com/OpenLoco/OpenLoco/assets/6529196/fa641acc-d33b-41dd-9e65-e811728c7eb3)
![image](https://github.com/OpenLoco/OpenLoco/assets/6529196/c8be75f9-485c-434b-9cb4-dbaae79bb7c4)

**Comment by AaronVanGeffen** [comment]
Yes, this looks like an integer overflow.

**Comment by superchadST345** [comment]
yes but it only seems to affect the Finances window and not the actual cash flow, at least for now

**Comment by AaronVanGeffen** [comment]
The finance window is using 32-bits numbers, while the actual cash flow is using 48-bit (!) numbers.

**Comment by superchadST345** [comment]
would it be possible to convert the finance Window to 48-bit numbers?

**Comment by AaronVanGeffen** [comment]
Of course. However, there are other priorities. Personally, I'd rather we go straight to 64-bit numbers when we can.
